hilti/toolchain/src/compiler/driver.cc

```diff
@@ -479,6 +490,9 @@ Result<Nothing> Driver::addInput(const hilti::rt::filesystem::path& path) {
     }
 
     else if ( path.extension() == ".cc" || path.extension() == ".cxx" ) {
+        if ( _driver_options.global_optimizations )
+            return result::Error("Loading C++ files is not supported with transformations enabled, rerun with '-g'");
+
```

Comment:
Iiirc, for this we arrived at allowing C++ to be added as long as they don't include HILTI linker information (because then they are probably just additional library code that won't have trouble with our optimizations).